python/lsst/verify/ap/verify_ap.py

```diff
@@ -0,0 +1,156 @@
+#
+# LSST Data Management System
+# Copyright 2017 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""Command-line program for running and analyzing AP pipeline.
+
+In addition to containing verify_ap's main function, this module manages
+command-line argument parsing.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__all__ = []
+
+import argparse
+import os
+import re
+
+import lsst.log
+from lsst.verify.ap.dataset import Dataset
+from lsst.verify.ap.metrics import MetricsParser, check_squash_ready, AutoJob
+from lsst.verify.ap.appipe import ApPipeParser, ApPipe
+
+
+class _VerifyApParser(argparse.ArgumentParser):
+    """An argument parser for data needed by this script.
+    """
+
+    def __init__(self):
+        argparse.ArgumentParser.__init__(
+            self,
+            description='Executes the LSST DM AP pipeline and analyzes its performance using metrics.',
+            epilog='',
+            parents=[ApPipeParser(), MetricsParser()],
+            add_help=True)
+        self.add_argument('--dataset', choices=Dataset.get_supported_datasets(), required=True,
+                          help='The source of data to pass through the pipeline.')
+
+        output = self.add_mutually_exclusive_group(required=True)
+        output.add_argument('--output', help='The location of the repository to use for program output.')
+        output.add_argument(
+            '--rerun', metavar='OUTPUT',
+            type=_FormattedType('[^:]+',
+                                'Invalid name "%s"; verify_ap does not support reruns as input.'),
```

Comment:
I think I'm still confused as to what the `type` argument is catching here. Normally you use it to clarify that the argparse input must be an int, str, float, whatever. But you're trying to catch some kind of user error where they mistakenly think `--rerun` could be used in lieu of `--dataset`?